internal/benchmarks/src/cost-estimation.bench.ts

```diff
@@ -1,36 +1,75 @@
-/* eslint-disable import/no-extraneous-dependencies */
-
-import type { TransferParams, Provider } from 'fuels';
-import { ScriptTransactionRequest, Wallet } from 'fuels';
-import { launchTestNode, TestAssetId } from 'fuels/test-utils';
-import { bench } from 'vitest';
+import { DEVNET_NETWORK_URL } from '@internal/utils';
+import type { TransferParams, TransactionCost } from 'fuels';
+import { Wallet, Provider, ScriptTransactionRequest, WalletUnlocked } from 'fuels';
+import { launchTestNode } from 'fuels/test-utils';
 
 import type { CallTestContract } from '../test/typegen/contracts';
 import { CallTestContractFactory } from '../test/typegen/contracts';
 
+import { isDevnet, runBenchmark } from './config';
+
 /**
  * @group node
  * @group browser
  */
 describe('Cost Estimation Benchmarks', () => {
   let contract: CallTestContract;
   let provider: Provider;
+  let request: ScriptTransactionRequest;
+  let recipient: WalletUnlocked;
+  let receiver1: WalletUnlocked;
+  let receiver2: WalletUnlocked;
+  let receiver3: WalletUnlocked;
+  let sender: WalletUnlocked;
   let cleanup: () => void;
-  beforeEach(async () => {
-    const launched = await launchTestNode({
-      contractsConfigs: [{ factory: CallTestContractFactory }],
-    });
-
-    cleanup = launched.cleanup;
-    contract = launched.contracts[0];
-    provider = contract.provider;
-  });
 
-  afterEach(() => {
-    cleanup();
+  const setupTestEnvironment = async () => {
+    if (isDevnet) {
+      provider = await Provider.create(DEVNET_NETWORK_URL);
+      const wallet = new WalletUnlocked(process.env.DEVNET_WALLET_PVT_KEY as string, provider);
+
+      const contractFactory = new CallTestContractFactory(wallet);
+      const { waitForResult } = await contractFactory.deploy<CallTestContract>();
+      const { contract: deployedContract } = await waitForResult();
+      contract = deployedContract;
+    } else {
+      const launched = await launchTestNode({
+        contractsConfigs: [{ factory: CallTestContractFactory }],
+      });
+
+      cleanup = launched.cleanup;
+      contract = launched.contracts[0];
+      provider = contract.provider;
+    }
+
+    request = new ScriptTransactionRequest({ gasLimit: 1000000 });
+    recipient = Wallet.generate({ provider });
+    receiver1 = Wallet.generate({ provider });
+    receiver2 = Wallet.generate({ provider });
+    receiver3 = Wallet.generate({ provider });
+    sender = Wallet.fromPrivateKey(
+      '0x30bb0bc68f5d2ec3b523cee5a65503031b40679d9c72280cd8088c2cfbc34e38',
```

Comment:
Whats the reason for this private key specifically?

Reply:
No reason, I've removed it be9151dc3